src/lib/hooks/use-floating.svelte.ts

```diff
@@ -0,0 +1,142 @@
+import { box } from '$lib/box.svelte.js';
+import type { UseFloatingOptions, UseFloatingReturn } from '$lib/types.js';
+import { getDPR, noop, roundByDPR, styleObjectToString } from '$lib/utils.js';
+import type { MiddlewareData, ReferenceElement } from '@floating-ui/dom';
+import { computePosition } from '@floating-ui/dom';
+import { onDestroy } from 'svelte';
+
+/**
+ * Hook for managing floating elements.
+ * Aims to keep as much parity with `@floating-ui/react` as possible.
+ * For now see: https://floating-ui.com/docs/useFloating for API documentation.
+ */
+export function useFloating<T extends ReferenceElement = ReferenceElement>(
+	options: UseFloatingOptions<T> = {}
+): UseFloatingReturn {
+	const openOption = box.derived(() => options.open ?? true);
+	const onOpenChangeOption = options.onOpenChange ?? noop;
+	const placementOption = box.derived(() => options.placement ?? 'bottom');
```

Comment:
Bit of a nitpick...

Since you have a lot of `xOption` and `xElement` names, it might be better to swap the verbiage around to `optionX` and `elemX` respectivly. This can help organize and group similar or related variables.

Ex:

```
optionFoo
optionBar
optionFizz

elemFoo
elemBar
elemFizz
```

Optionally, maybe consider wrapping with an object to make this relationship more explicit (assuming it doesn't impact reactivity).

Reply:
Yeah grouping them would influence reactivity. About the naming suggestion, I have to disagree here, although it makes grouping easier. It doesn't read "properly".
What is this variable? Oh it's the `placementOption`
What is this variable? Oh it's the `optionPlacement`
The latter doesn't read like english. And variables are supposed to make things more clear.